### PR TITLE
test: add test for iroh-dns-server with mainline

### PR DESCRIPTION
**Pull request by Frando**
## Description

This adds a test for iroh-dns-server with mainline fallback resolution.
Needed to rework the config structure a little to not duplicate code too much.

## Breaking Changes

are documented in #2180 

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [x] Tests if relevant.
- [x] All breaking changes documented.

### CODE REVIEW

File: iroh-dns-server/src/config.rs [hunk 1]
```diff
@@ -69,12 +69,19 @@ impl MetricsConfig {
 pub struct MainlineConfig {
     /// Set to true to enable the mainline lookup.
     pub enabled: bool,
+    /// Set custom bootstrap nodes.
+    ///
+    /// If empty this will use the default bittorrent mainline bootstrap nodes as defined by pkarr.
+    pub bootstrap: Vec<SocketAddr>,
```
Comment (rklaehn) [inline iroh-dns-server/src/config.rs:75]: OK, glad we got something else to put into the mainline config section...

File: iroh-dns-server/src/config.rs [hunk 2]
```diff
@@ -69,12 +69,19 @@ impl MetricsConfig {
...
 }
 
 #[allow(clippy::derivable_impls)]
 impl Default for MainlineConfig {
     fn default() -> Self {
-        Self { enabled: false }
+        Self {
+            enabled: false,
+            bootstrap: vec![],
```
Comment (rklaehn) [inline iroh-dns-server/src/config.rs:83]: Empty means default? How do you get the default mainline bootstrap nodes?

File: iroh-dns-server/src/store.rs [hunk 3]
```diff
@@ -55,12 +58,26 @@ impl ZoneStore {
...
+    ///
+    /// Optionally set custom bootstrap nodes. If `bootstrap` is empty it will use the default
+    /// mainline bootstrap nodes.
+    pub fn with_mainline_fallback(self, bootstrap: &[SocketAddr]) -> Self {
+        let pkarr_client = if bootstrap.is_empty() {
+            PkarrClient::default()
+        } else {
+            let bootstrap = bootstrap
+                .iter()
+                .map(|addr| addr.to_string())
```
Comment (rklaehn) [inline iroh-dns-server/src/store.rs:73]: Can these be URLs, or why does pkarr require strings here? Hopefully that can be changed in pkarr 2.0

File: iroh-dns-server/src/config.rs (hunk 2 above)
Comment (rklaehn) [reply iroh-dns-server/src/config.rs:83]: I know it is annoying, but can we make this an enum Default/Custom instead of using empty as "use the defaults"?

**Review by rklaehn — APPROVED** [review]

File: iroh-dns-server/src/store.rs (hunk 3 above)
Comment (Frando) [reply iroh-dns-server/src/store.rs]: Good point. I misread the pkarr source - these can be either "ip:port" or "domain:port" strings. changed to use strings on our end as well.

**Pull request merged**